extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultHealthCheckTest.java

```diff
@@ -9,67 +9,109 @@
 
-        assertThat(result).isSucceeded();
+            var result = healthCheck.get();
+
+            assertThat(result).isSucceeded();
+        }
+
+        @Test
+        void get_whenHealthCheckFailed_shouldFail() {
+            var healthCheckErr = "Vault is not available. Reason: Vault is in standby, additional information: hello";
+            when(client.doHealthCheck()).thenReturn(Result.failure(healthCheckErr));
+
+            var result = healthCheck.get();
+
+            assertThat(result).isFailed();
+            assertFailureMessagesSize(result, 1);
+            verify(monitor).warning("Healthcheck failed with reason(s): %s".formatted(healthCheckErr));
+        }
     }
 
-    @ParameterizedTest
-    @ValueSource(ints = {409, 472, 473, 501, 503, 999})
-    void shouldFail_whenClientReturnsErrorCodes(int code) {
-        var response = HealthResponse.Builder.newInstance().payload(new HealthResponsePayload()).code(code).build();
-        when(client.getHealth()).thenReturn(response);
+    @Nested
+    class HealthCheck200 {
 
-        var result = healthCheck.get();
+        @BeforeEach
+        void beforeEach() {
+            when(client.doHealthCheck()).thenReturn(Result.success());
+        }
 
-        assertThat(result).isFailed();
-        verify(monitor, times(1)).warning(anyString());
+        @Test
+        void get_whenTokenValid_shouldSucceed() {
+            when(client.lookUpToken()).thenReturn(TOKEN_LOOK_UP_RESULT_200);
+
+            var result = healthCheck.get();
+
+            assertThat(result).isSucceeded();
+        }
+
+        @Test
+        void get_whenTokenNotValid_shouldFail() {
+            var tokenLookUpErr = "Token look up failed with status 403";
+            when(client.lookUpToken()).thenReturn(Result.failure(tokenLookUpErr));
+
+            var result = healthCheck.get();
+
+            assertThat(result).isFailed();
+            assertFailureMessagesSize(result, 1);
+            verify(monitor).warning("Healthcheck failed with reason(s): %s".formatted(tokenLookUpErr));
+        }
     }
 
     @Test
-    void testResponseFromException() {
-        when(client.getHealth()).thenThrow(new EdcException("foo-bar"));
+    void get_whenHealthCheckFailedAndTokenNotValid_shouldFail() {
+        var healthCheckErr = "Vault is not available. Reason: Vault is in standby, additional information: hello";
+        var tokenLookUpErr = "Token look up failed with status 403";
+
+        when(client.doHealthCheck()).thenReturn(Result.failure(healthCheckErr));
+        when(client.lookUpToken()).thenReturn(Result.failure(tokenLookUpErr));
 
         var result = healthCheck.get();
 
         assertThat(result).isFailed();
+        assertFailureMessagesSize(result, 2);
+        verify(monitor).warning("Healthcheck failed with reason(s): %s, %s".formatted(healthCheckErr, tokenLookUpErr));
+    }
+
+    private void assertFailureMessagesSize(HealthCheckResult result, int i) {
```

Comment:
personally, i really dislike these custom "assert-something" methods, especially if the assertion is as trivial as it is here. They make the code less readable. pls inline.